base/log_keys.go

```diff
@@ -64,16 +67,46 @@ func (keyMask *LogKey) Disable(logKey LogKey) {
 	atomic.StoreUint32((*uint32)(keyMask), val & ^uint32(logKey))
 }
 
-// Enabled returns true if the given logKey, or KEY_ALL is enabled in keyMask.
+// Enabled returns true if the given logKey is enabled in keyMask.
+// Always returns true if KEY_ALL is enabled in keyMask.
+// Always returns false if KEY_NONE is enabled in keyMask.
 func (keyMask *LogKey) Enabled(logKey LogKey) bool {
 	return keyMask.enabled(logKey, true)
 }
 
 // enabled returns true if the given logKey is enabled in keyMask, with an optional wildcard check.
-func (keyMask *LogKey) enabled(logKey LogKey, checkWildcard bool) bool {
+func (keyMask *LogKey) enabled(logKey LogKey, checkWildcards bool) bool {
+	if keyMask == nil {
+		return false
+	}
+
 	flag := atomic.LoadUint32((*uint32)(keyMask))
-	return (checkWildcard && flag&uint32(KEY_ALL) != 0) ||
-		flag&uint32(logKey) != 0
+
+	if checkWildcards {
+		// If KEY_NONE is set, return false for everything.
+		if flag&uint32(KEY_NONE) != 0 {
+			return false
+		}
+		// If KEY_ALL is set, return true for everything.
+		if flag&uint32(KEY_ALL) != 0 {
+			return true
+		}
+	}
+
+	return flag&uint32(logKey) != 0
+}
+
+func (keyMask *LogKey) MarshalText() ([]byte, error) {
+	if keyMask == nil {
+		return nil, errors.New("invalid log key")
+	}
+	return []byte(LogKeyName(*keyMask)), nil
+}
```

Comment:
Since LogKey is a bitfield, text marshalling based on a single LogKeyName doesn't seem correct.  Should this be returning a string built from keyMask.EnabledLogKeys?

Reply:
Actually Marshal/UnmarshalText are unused since I changed to use a `[]string` in the config, rather than `string`. I'll remove these completely. Thanks.